ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ConnectorUtils.java

```diff
@@ -358,6 +360,32 @@ public static SdkHttpFullRequest buildSdkRequest(
         return builder.build();
     }
 
+    public static Request buildOKHttpRequestPOST(String action, Connector connector, Map<String, String> parameters, String payload) {
+        okhttp3.RequestBody requestBody;
+        if (payload != null) {
+            requestBody = okhttp3.RequestBody.create(payload, MediaType.parse("application/json; charset=utf-8"));
+        } else {
+            throw new IllegalArgumentException("Content length is 0. Aborting request to remote model");
```

Comment:
What happens after that? What does customer see finally?

Reply:
This is not a helpful message and also this is a 4xx error. At this point as customer I won't have any idea what to do to fix this issue.